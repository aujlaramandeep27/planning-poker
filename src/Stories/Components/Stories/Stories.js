import React, {useEffect} from 'react';
import {viewCreateStory, viewMeeting, viewMeetings, viewStory} from '../../../Navigation/route-actions';
import {Page} from '../../../Common/Page';
import {connect} from 'react-redux';
import {DeleteStory, GetStories} from "../../Actions/StoriesActions";

export function Stories({currentMeeting, stories = [], goToMeetings, goToStory, deleteStory, createStory, goToEstimate, getStories, match}) {

    useEffect(
        () => {
            getStories(match.params.meetingId);
        },
        [getStories, match]
    );

    return (
        <Page title={currentMeeting.name} onBack={goToMeetings}>
            <button className="uk-button uk-button-primary uk-button-small uk-position-small uk-position-top-right"
                    onClick={() => goToEstimate(currentMeeting._id)}>
                Estimate
            </button>

            {stories.map((story, index) =>
                <div key={index} className="uk-card uk-card-primary uk-card-body uk-margin-small">
                    <div className="uk-card-badge uk-label">Points: 3</div>
                    <h3 className="uk-card-title" onClick={() => goToStory(currentMeeting._id, story._id)}>
                        <a className="uk-link-heading">{story.name}</a>
                    </h3>
                    <button
                        className="uk-position-center-right uk-margin-small-right"
                        data-uk-icon="icon: trash"
                        onClick={() => RemoveStory(story._id)}>
                    </button>
                </div>
            )}

            <div className="uk-card uk-card-secondary uk-card-body" onClick={createStory}>
                <h3 className="uk-card-title">+</h3>
            </div>
        </Page>
    );

    function RemoveStory(storyId) {
        deleteStory(storyId);
    }
}

function mapStateToProps(state) {
    return {
        currentMeeting: state.currentMeeting,
        stories: state.stories,
        error: state.error
    };
}

function mapDispatchToProps(dispatch) {
    return {
        goToMeetings: () => dispatch(viewMeetings()),
        goToStory: (meetingId, storyId) => dispatch(viewStory(meetingId, storyId)),
        deleteStory: (storyId) => dispatch(DeleteStory(storyId)),
        createStory: () => dispatch(viewCreateStory()),
        goToEstimate: (meetingId) => dispatch(viewMeeting(meetingId)),
        getStories: (meetingId) => dispatch(GetStories(meetingId))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Stories);
