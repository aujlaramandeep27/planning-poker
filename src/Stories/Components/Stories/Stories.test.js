import {Stories} from './Stories';
import {cleanup, render} from "@testing-library/react";
import React from "react";
describe('Container: Story', ()=> {

    afterEach(cleanup);

    it('should render correctly', () => {

        const props = {
            currentMeeting: jest.fn().mockReturnValue({
                name: 'Mock Meeting Name'
            }),
            stories: [{
                id: 'id',
                name: 'Mock Story Name'
            }],
            error: null,
            goToMeetings: jest.fn(),
            goToStory: jest.fn(),
            deleteStory: jest.fn(),
            createStory: jest.fn(),
            goToEstimate: jest.fn(),
            getStories: jest.fn()
        };
        const {container} = render(<Stories  {...props} match={{'params': {'meetingId': 'Mock meeting id'}}}/>);
        expect(container).toMatchSnapshot();
    });
});
