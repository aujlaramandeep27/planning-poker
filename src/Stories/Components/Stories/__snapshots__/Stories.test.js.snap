// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories renders correctly 1`] = `
<div>
  <div
    class="default-background"
  >
    <div
      class="uk-container uk-text-center@m"
    >
      <h1
        class="uk-heading-divider uk-margin-top"
      >
        mockConstructor
      </h1>
      <div
        class="uk-align-center uk-width-1-2@m"
      >
        <span
          class="uk-position-large uk-position-top-left"
          data-uk-icon="icon: arrow-left; ratio: 3"
        />
        <button
          class="uk-button uk-button-primary uk-button-small uk-position-small uk-position-top-right"
        >
          Estimate
        </button>
        <div
          class="uk-card uk-card-primary uk-card-body uk-margin-small"
        >
          <div
            class="uk-position-center-left uk-margin-small-left uk-label"
          >
            Points: 3
          </div>
          Mock Story Name
          <button
            class="uk-position-center-right uk-margin-small-right"
            data-uk-icon="icon: trash"
          />
        </div>
        <div
          class="uk-card uk-card-secondary uk-card-body"
        >
          <h3
            class="uk-card-title"
          >
            +
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
